### PR TITLE
Change XML and Configuration Loading

### PR DESCRIPTION
**Pull request by BrentHoltsclaw**
#741

### CODE REVIEW

File: chipsec/chipset.py [hunk 1]
```diff
@@ -606,24 +283,92 @@ def print_pch(self):
...
+                        mdict = self.chipset_dictionary[int(vid,16)]
+                        cdict = self.chipset_codes[_cfg.attrib['platform'].lower()]
+                    else: continue
+                    if logger().DEBUG: logger().log( "[*] Populating configuration dictionary.." )
+                    for _info in _cfg.iter('info'):
+                        if 'family' in _info.attrib:
+                            if _info.attrib['family'].lower() == "core":
+                                CHIPSET_FAMILY_CORE.append(_cfg.attrib['platform'].lower())
+                            if _info.attrib['family'].lower() == "atom":
+                                CHIPSET_FAMILY_XEON.append(_cfg.attrib['platform'].lower())
```
Comment (NicholasArmour) [inline chipsec/chipset.py:357]: you have swapped xeon and atom...

Comment (BrentHoltsclaw) [reply chipsec/chipset.py:357]: Fixed in #871.  I can fix it here too if necessary

File: chipsec/cfg/8086/kbl.xml [hunk 2]
```diff
@@ -10,6 +10,23 @@ http://www.intel.com/content/www/us/en/processors/core/core-technical-resources.
...
+  <info family="core">
+    <sku did="0x5900" name="Kabylake" code="KBL" longname="Mobile 7th Generation Core Processor (Kabylake H)" />
+    <sku did="0x5904" name="Kabylake" code="KBL" longname="Mobile 7th Generation Core Processor (Kabylake U)" />
+    <sku did="0x590C" name="Kabylake" code="KBL" longname="Mobile 7th Generation Core Processor (Kabylake Y)" />
+    <sku did="0x590F" name="Kabylake" code="KBL" longname="Desktop 7th Generation Core Processor (Kabylake S)" />
+    <sku did="0x5910" name="Kabylake" code="KBL" longname="Mobile 7th Generation Core Processor (Kabylake H)" />
+    <sku did="0x5914" name="Kabylake" code="KBL" longname="Mobile 8th Generation Core Processor (Kabylake U-Quad Core)" />
+    <sku did="0x591F" name="Kabylake" code="KBL" longname="Desktop 7th Generation Core Processor (Kabylake S)" />
+    <sku did="0x5918" name="Kabylake" code="KBL" longname="Mobile 7th Generation Core Processor (Kabylake H)" />
+    <sku did="0x5918" name="Kabylake" code="KBL" longname="Intel Xeon Processor E3 v6 (Kabylake CPU)" />
```
Comment (frinzell) [inline chipsec/cfg/8086/kbl.xml:27]: Duplicate DID.

File: chipsec/cfg/8086/pch_3xxop.xml [hunk 3]
```diff
@@ -5,6 +5,16 @@ XML configuration file for the 300 series On Package PCH
...
 -->
+
+  <!-- #################################### -->
+  <!--                                      -->
+  <!-- Information                          -->
+  <!--                                      -->
+  <!-- #################################### -->
+  <info>
+    <sku did="0x9D84" name="PCH-U" code="PCH_3xx" longname="Intel 300 series On-Package PCH" />
+  </info>
```
Comment (frinzell) [inline chipsec/cfg/8086/pch_3xxop.xml:16]: Should this be `code = "PCH_3xxOP"`?

File: chipsec/cfg/8086/pch_3xxlp.xml [hunk 4]
```diff
@@ -5,6 +5,21 @@ XML configuration file for the 300 series LP (U/Y) PCH
...
+  <!--                                      -->
+  <!-- #################################### -->
+  <info>
+    <sku did="0x9D4B" name="PCH-U"  code="3xxLP" longname="PCH-Y with iHDCP 2.2 Premium" />
+    <sku did="0x9D4E" name="PCH-U"  code="3xxLP" longname="PCH-U with iHDCP 2.2 Premium" />
+    <sku did="0x9D50" name="PCH-U"  code="3xxLP" longname="PCH-U with iHDCP 2.2 Base" />
+    <sku did="0x9D53" name="PCH-U"  code="3xxLP" longname="PCH-U Base" />
+    <sku did="0x9D56" name="PCH-Y"  code="3xxLP" longname="PCH-Y Premium" />
+    <sku did="0x9D58" name="PCH-U"  code="3xxLP" longname="PCH-U Premium" />
+  </info>
```
Comment (frinzell) [inline chipsec/cfg/8086/pch_3xxlp.xml:21]: Should all of these be `code="PCH_3xxLP"`?

File: chipsec/cfg/8086/pch_c60x.xml [hunk 5]
```diff
@@ -7,6 +7,15 @@ XML configuration file for C600 series PCH
...
 -->
 
+  <!-- #################################### -->
+  <!--                                      -->
+  <!-- Information                          -->
+  <!--                                      -->
+  <!-- #################################### -->
+  <info>
+    <sku did="0x1D41" name="C600" code="PCH_C60x" longname="Intel C600/X79 series PCH" />
+  </info>
```
Comment (frinzell) [inline chipsec/cfg/8086/pch_c60x.xml:17]: Are `code` and `platform` case sensitive?  The 'x' is lower case.

File: chipsec/module_common.py [hunk 6]
```diff
@@ -72,7 +72,6 @@ def is_supported(self):
         platform.
         To access the currently running platform use
 
-        >>> self.cs.get_chipset_id()
         """
```
Comment (frinzell) [inline chipsec/module_common.py:75]: Copyright in this file

File: chipsec/utilcmd/chipset_cmd.py [hunk 7]
```diff
@@ -27,7 +27,7 @@
 """
 
 from chipsec.command    import BaseCommand
-from chipsec.chipset    import UnknownChipsetError, print_supported_chipsets
+from chipsec.chipset    import UnknownChipsetError
```
Comment (frinzell) [inline chipsec/utilcmd/chipset_cmd.py:30]: Copyright on this file

File: chipsec/cfg/8086/pch_c61x.xml [hunk 8]
```diff
@@ -6,6 +6,17 @@ XML configuration file for C610 series PCH
...
+  <!-- #################################### -->
+  <!--                                      -->
+  <!-- Information                          -->
+  <!--                                      -->
+  <!-- #################################### -->
+  <info>
+    <sku did="0x8D40" name="C610" code="PCH_C61x" longname="Intel Wellsburg (C610/X99 series) PCH" />
+    <sku did="0x8D44" name="C610-G" code="PCH_C61x" longname="Intel Wellsburg-G (C610/X99 series) PCH" />
+    <sku did="0x8D47" name="C610-X" code="PCH_C61x" longname="Intel Wellsburg-X (C610/X99 series) PCH" />
+  </info>
```
Comment (frinzell) [inline chipsec/cfg/8086/pch_c61x.xml:18]: Safer to capitalize X?

File: chipsec/cfg/8086/whl.xml [hunk 9]
```diff
@@ -8,6 +8,15 @@ XML configuration file for Whiskey Lake
...
 -->
 
+  <!-- #################################### -->
+  <!--                                      -->
+  <!-- Information                          -->
+  <!--                                      -->
+  <!-- #################################### -->
+  <info family="core">
+    <sku did="0x3E34" name="Whiskey Lake" code="WHL" longname="Mobile 8th Generation Core Processor (Whiskey Lake U 4 Cores)" detection_value="806EA" />
+  </info>
```
Comment (frinzell) [inline chipsec/cfg/8086/whl.xml:18]: Missing the '806EB' entry.

File: chipsec/cfg/8086/dnv.xml [hunk 10]
```diff
@@ -8,12 +8,15 @@ XML configuration file for Denverton
...
+    <sku did="0x1980" name="Denverton" code="DNV" longname="Intel Atom Processor C3000 Product Family" />
+  </info>
+
   <pci>
   </pci>
 
-
   <mmio>
-    <bar name="SPIBAR"   bus="0" dev="0x1F" fun="5" reg="0x10" width="4" mask="0xFFFFC000" size="0x200"  desc="SPI Controller Register Range"/> 
   </mmio>
```
[the remark this answers is not in the document]
Comment (BrentHoltsclaw) [reply chipsec/cfg/8086/dnv.xml:20]: Not sure fixing

File: chipsec/cfg/8086/pch_3xxop.xml (hunk 3 above)
Comment (BrentHoltsclaw) [reply chipsec/cfg/8086/pch_3xxop.xml:16]: fixing

File: chipsec/cfg/8086/pch_c60x.xml (hunk 5 above)
Comment (BrentHoltsclaw) [reply chipsec/cfg/8086/pch_c60x.xml:17]: they aren't but for consistency will make them match

File: chipsec/chipset.py [hunk 11]
```diff
@@ -503,65 +188,64 @@ def get_cpuid(self):
...
             else:
                 self.pch_vid = 0xFFFF
                 self.pch_did = 0xFFFF
                 self.pch_rid = 0xFF
 
-        if self.pch_vid == VID_INTEL and self.pch_did in pch_dictionary:
-            _unknown_pch = False
-            data_dict           = pch_dictionary[self.pch_did][0]
+        if self.pch_vid in self.pch_dictionary.keys() and self.pch_did in self.pch_dictionary[self.pch_vid].keys():
+            data_dict           = self.pch_dictionary[self.vid][self.pch_did][0]
```
Comment (npmitche) [inline chipsec/chipset.py:245]: Removed _unknown_pch. Will error on line 259 when checking undefined variable.

**Pull request merged**